compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -10259,7 +10259,13 @@ TR::Node *ishlSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier * s)
    if (identity)
       return identity;
 
-   if (secondChild->getOpCode().isLoadConst() &&
+   // Replace shift of constant zero with constant zero
+   if (firstChild->getOpCode().isLoadConst() && firstChild->isZero() &&
```

Comment:
I'll have to take a closer look, but I think that `isZero()` usually won't return true until after Value Propagation has run.  May I ask you to check that?  If I'm correct, it would probably be better to check the value returned by `firstChild->getInt()` (or `getShort()`, _et al._, as appropriate), so that this optimization doesn't rely on Value Propagation already having run.

Reply:
I did see calls to `setIsZero` during value propagation while I was working on this, I just sort of assumed that simplification always happens after value propagation because I saw them happen in that order in a log. I'll fix that!